Common/MathUtils/src/Chebyshev3D.cxx

```diff
@@ -507,15 +507,15 @@ Int_t Chebyshev3D::chebyshevFit(int dmOut)
 {
   // prepare paramaterization of 3D function for dmOut-th dimension
   int maxDim = 0;
-  for (int i = 0; i < 3; i++) {
-    if (maxDim < mNumberOfPoints[i]) {
-      maxDim = mNumberOfPoints[i];
+  for (int mNumberOfPoint : mNumberOfPoints) {
+    if (maxDim < mNumberOfPoint) {
```

Comment:
Agreed on the variable name, so let's hold for this, as I said. Having an index is in general error prone after modifications, not to start with. In this case if you were to change mNumberOfPoints from size X to size Y, you would not need to adapt your code with the refactored loop, while you most likely would if the index was explicitly mentioned in the for loop.